sys/parsoid.js

```diff
@@ -936,12 +936,21 @@ class ParsoidService {
 
     _getOriginalContent(hyper, req, revision, tid) {
         const rp = req.params;
-        return this._getContentWithFallback(hyper, rp.domain, rp.title, revision, tid)
-            .then((res) => {
-                res = res.body;
-                res.revid = revision;
-                return res;
-            });
+
+        const disabledStorage = this._isStorageDisabled(req.params.domain);
+        let contentReq;
+
+        if (disabledStorage) {
+            contentReq = this._getPageBundleFromParsoid(hyper, req);
```

Comment:
Can you confirm that this will fetch a stashed version from core? Otherwise, this could lead to dirty diffs? How are you testing this?

Reply:
Disabling storage also disabeld stashing, since RESTbase uses the storage mechanism to implement the stash. It doesn't use the Stash in MW.

But that should be a non-issue, since VE doesn't use this API anymore. As far as I know, nothing uses stashing in RESTbase. I have a PR up for removing it entirely.